### ISSUE TITLE
Issue with rendering to a RenderWindow

### ISSUE THREAD

**Issue by CodeAX2** [SFML/SFML#1525]
Issue with rendering to a RenderWindow

I have been working on a top-down 2D game for a while now. It seems as though when I render tiles in the game, SFML tends to have some issues with rendering them. Often, the texture of the tile gets set to a single color, that typically corresponds to the correct texture of the tile. I have looked everywhere in my own code to attempt to find what is causing this, to no avail. Others have looked at my code as well, and have also concluded that this is most likely some bug within SFML itself.

My OS is Windows 10
SFML Version 2.5.0
Compiler is Visual Studio 2017

The important snippet of my code can be found here: https://pastebin.com/LNgzCb61
Since I believe the issue to be caused by trying to render a large amount of different sf::Textures to the window, and requires a lot of code to reproduce.

The expected behavior is that the sf::RectangleShapes will render with their correct textures, and not as a single color

The actual behavior is the the shapes are rendered with a solid color, a color that corresponds to the actual texture of the rectangle (i.e. water is solid blue). This only happens for 1 frame, as on the next frame the tile is back to its normal texture.

I have spent countless hours working to try and find an error in my code that would cause this issue, so it is fairly likely that this is an SFML issue.

In addition an example of the bug in action can be found here: https://www.youtube.com/watch?v=93n_V0vjkHI&feature=youtu.be

**Comment by eXpl0it3r** [comment]
Sounds like the common [White Square problem](https://www.sfml-dev.org/tutorials/2.5/graphics-sprite.php#the-white-square-problem). But without minimal and complete example it's impossible to tell what's really going on.

If you manage Textures in a vector, then all textures pointers will get invalidated when you push_back new texture object.

You can debug your code and check what happens to the texture pointers inside your shapes, that way, you'll see when things go wrong.

Btw we prefer to discuss such problems on the forum and use this issue tracker for confirmed bugs.

**Comment by CodeAX2** [comment]
I tried to leave as much code as I could. I’ll work on replicating the bug with some more code. This is definitely not the white square problem, as the texture pointers are never null. I am also not using a vector for my textures, I’m just using an array. I create my textures with new, so that they don’t go away until I delete them at the end of my program. And like I said, I do believe this to be a bug inside of SFML, since me, and many others for that matter, have looked all over my code, and have not found any spot that would lead to this issue.

**Comment by eXpl0it3r** [comment]
It might certainly an issue with SFML, if it is, then the issue can be reproduced with a minimal example.

**Comment by CodeAX2** [comment]
Yeah, I understand that my code wasn't very minimal or complete, but it was unfortunately all that I can really give out, without just pasting my entire project (which would not be minimal). I tried to make it as simple as I could, but it seems that this problem is fairly complicated. And after even more hours of basically trying random stuff, I think I found out what was truly happening. OpenGL was bugging out, since I was just using the default version, being 1.1. Instead, I set it to be 4.4, which removed the bug completely from what I can tell! Thankfully it wasn't a bug in either my code OR SFML, and was just simply OpenGL messing with me (from what I can tell at least). So, sorry for all the fuss, I just wanted to post this in case others were experiencing a similar issue, and because it had nothing to do with my own code, I figured it was SFML, when in reality it was OpenGL.

Thanks for the responses and help though! I'm glad that SFML has such a great team behind it
